ocaml/libs/stunnel/stunnel.ml

```diff
@@ -287,6 +296,50 @@ let disconnect ?(wait = true) ?(force = false) x =
    exception instead *)
 exception Stunnel_initialisation_failed
 
+let with_client_config_file ~config f =
+  let tmpfile, tmpch =
+    Filename.open_temp_file ~mode:[Open_text] "stunnel-client-proxy-" ".conf"
+  in
+  Xapi_stdext_pervasives.Pervasiveext.finally
+    (fun () -> output_string tmpch config ; close_out tmpch ; f tmpfile)
+    (fun () -> Unixext.unlink_safe tmpfile)
+
+let with_client_proxy verify_cert r_host r_port l_host l_port f =
+  let config =
+    config_file ~accept:(Some (l_host, l_port)) verify_cert r_host r_port
+  in
+  with_client_config_file ~config @@ fun config_file_path ->
```

Comment:
It's possible. But my first impression was it might increase the number of branches in the `with_attempt_one_connect`. Lots of bits in the function are handling the socketpair which is not necessary in this task.
But I would like to have a try to see what it would be in reusing way.